// @rjsf/semantic-ui/dist/cjs/TitleField/TitleField.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _semanticUiReact = require("semantic-ui-react");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function TitleField(_ref) {
  var title = _ref.title,
      options = _ref.options;
  var semantic = options.semantic;

  if (title) {
    return _react.default.createElement(_semanticUiReact.Header, _extends({}, semantic, {
      as: "h5"
    }), title);
  }
}

TitleField.defaultProps = {
  options: {
    semantic: {
      inverted: false,
      dividing: true
    }
  }
};
TitleField.propTypes = {
  options: _propTypes.default.object
};
var _default = TitleField;
exports.default = _default;
