// @rjsf/semantic-ui/dist/cjs/FieldTemplate/FieldTemplate.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _semanticUiReact = require("semantic-ui-react");

var _DescriptionField = _interopRequireDefault(require("../DescriptionField"));

var _HelpField = _interopRequireDefault(require("../HelpField"));

var _RawErrors = _interopRequireDefault(require("../RawErrors"));

var _util = require("../util");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function FieldTemplate(_ref) {
  var id = _ref.id,
      children = _ref.children,
      className = _ref.className,
      classNames = _ref.classNames,
      displayLabel = _ref.displayLabel,
      label = _ref.label,
      _ref$rawErrors = _ref.rawErrors,
      rawErrors = _ref$rawErrors === void 0 ? [] : _ref$rawErrors,
      rawHelp = _ref.rawHelp,
      rawDescription = _ref.rawDescription,
      props = _objectWithoutProperties(_ref, ["id", "children", "className", "classNames", "displayLabel", "label", "rawErrors", "rawHelp", "rawDescription"]);

  var semanticProps = (0, _util.getSemanticProps)(props);
  var wrapLabel = semanticProps.wrapLabel,
      wrapContent = semanticProps.wrapContent,
      errorOptions = semanticProps.errorOptions;
  return _react.default.createElement(_semanticUiReact.Form.Group, {
    key: id,
    widths: "equal",
    grouped: true
  }, _react.default.createElement(_util.MaybeWrap, {
    wrap: wrapContent,
    className: "sui-field-content"
  }, children, displayLabel && rawDescription && _react.default.createElement(_util.MaybeWrap, {
    wrap: wrapLabel,
    className: "sui-field-label"
  }, rawDescription && _react.default.createElement(_DescriptionField.default, {
    description: rawDescription
  })), _react.default.createElement(_HelpField.default, {
    helpText: rawHelp,
    id: id + "__help"
  }), _react.default.createElement(_RawErrors.default, {
    errors: rawErrors,
    options: errorOptions
  })));
}

var _default = FieldTemplate;
exports.default = _default;
