// @rjsf/semantic-ui/dist/cjs/RawErrors/RawErrors.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _nanoid = require("nanoid");

var _semanticUiReact = require("semantic-ui-react");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/* eslint-disable react/no-array-index-key */

/**
 *
 * @param errors
 * @param displayError
 * @returns {*}
 * @constructor
 * @return {null}
 */
function RawErrors(_ref) {
  var errors = _ref.errors,
      options = _ref.options;
  var pointing = options.pointing,
      size = options.size;

  if (errors && errors.length > 0) {
    return _react.default.createElement(_semanticUiReact.Label, {
      color: "red",
      pointing: pointing || "above",
      size: size || "small",
      basic: true
    }, _react.default.createElement(_semanticUiReact.List, {
      bulleted: true
    }, errors.map(function (error) {
      return _react.default.createElement(_semanticUiReact.List.Item, {
        key: (0, _nanoid.nanoid)(),
        content: error
      });
    })));
  }

  return null;
}

RawErrors.defaultProps = {
  options: {
    pointing: "above",
    size: "small"
  }
};
RawErrors.propTypes = {
  options: _propTypes.default.object,
  errors: _propTypes.default.array
};
var _default = RawErrors;
exports.default = _default;
